install/src/Stack.hs

```diff
@@ -102,7 +104,9 @@ stackBuildFailMsg =
 -- |Run actions without the stack cached binaries 
 withoutStackCachedBinaries :: Action a -> Action a
 withoutStackCachedBinaries action = do
-  mbPath <- liftIO (lookupEnv "PATH")
+  mbPath <- liftIO (catch
+                    (lookupEnv "PATH")
+                    (\(_ :: SomeException) -> return Nothing))
```

Comment:
`SomeException` also catches Async Exceptions. In this case, this may be safe, but it is a bad idea in general.
Please just catch `IOException` (assuming that is the actual exception that is thrown.)

Reply:
Oh, right, thanks for pointed it out